UnauthorizedAccessException for apphost.exe when trying to use latest nightly dotnet build

When I try to use `dotnet build` or `dotnet run` with the latest nightly build, it crashes with this error:
"System.UnauthorizedAccessException: Access to the path 'C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64\6.0.0-preview.3.21156.6\runtimes\win-x64\native\apphost.exe' is denied."
```
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018: The "CreateAppHost" task failed unexpectedly. [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018: System.UnauthorizedAccessException: Access to the path 'C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64\6.0.0-preview.3.21156.6\runtimes\win-x64\native\apphost.exe' is denied. [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at System.IO.FileStreamHelpers.ValidateFileHandle(SafeFileHandle fileHandle, String path, Boolean useAsyncIO) [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at System.IO.FileStreamHelpers.CreateFileOpenHandle(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options) [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at System.IO.LegacyFileStreamStrategy..ctor(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options) [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at System.IO.MemoryMappedFiles.MemoryMappedFile.CreateFromFile(String path, FileMode mode, String mapName, Int64 capacity, MemoryMappedFileAccess access) [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at Microsoft.NET.HostModel.AppHost.HostWriter.<>c__DisplayClass2_0.<CreateAppHost>b__2() [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at Microsoft.NET.HostModel.RetryUtil.RetryOnIOError(Action func) [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at Microsoft.NET.HostModel.AppHost.HostWriter.CreateAppHost(String appHostSourceFilePath, String appHostDestinationFilePath, String appBinaryFilePath, Boolean windowsGraphicalUserInterface, String assemblyToCopyResorcesFrom) [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at Microsoft.NET.Build.Tasks.CreateAppHost.ExecuteCore() [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.3.21158.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(522,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [C:\users\stoub\Desktop\Benchmarks\Benchmarks.csproj]
```

--- a/src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs
+++ b/src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs
@@ -7,7 +7,6 @@ using System.IO;
 using System.IO.MemoryMappedFiles;
 using System.Runtime.InteropServices;
 using System.Text;
-using System.Threading;
 
 namespace Microsoft.NET.HostModel.AppHost
 {
@@ -91,7 +90,7 @@ namespace Microsoft.NET.HostModel.AppHost
                     try
                     {
                         // Open the source host file.
-                        memoryMappedFile = MemoryMappedFile.CreateFromFile(appHostSourceFilePath, FileMode.Open, null, 0, MemoryMappedFileAccess.CopyOnWrite);
+                        memoryMappedFile = MemoryMappedFile.CreateFromFile(appHostSourceFilePath, FileMode.Open, null, 0, MemoryMappedFileAccess.Read);
                         memoryMappedViewAccessor = memoryMappedFile.CreateViewAccessor(0, 0, MemoryMappedFileAccess.CopyOnWrite);
 
                         // Transform the host file in-memory.

--- a/src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs
+++ b/src/installer/tests/Microsoft.NET.HostModel.Tests/Microsoft.NET.HostModel.AppHost.Tests/AppHostUpdateTests.cs
@@ -201,6 +201,25 @@ namespace Microsoft.NET.HostModel.Tests
             int GetLastError() => Marshal.GetLastWin32Error();
         }
 
+        [Fact]
+        public void CanCreateAppHost()
+        {
+            using (TestDirectory testDirectory = TestDirectory.Create())
+            {
+                string sourceAppHostMock = PrepareAppHostMockFile(testDirectory);
+                File.SetAttributes(sourceAppHostMock, FileAttributes.ReadOnly);
+                string destinationFilePath = Path.Combine(testDirectory.Path, "DestinationAppHost.exe.mock");
+                string appBinaryFilePath = "Test/App/Binary/Path.dll";
+                HostWriter.CreateAppHost(
+                   sourceAppHostMock,
+                   destinationFilePath,
+                   appBinaryFilePath,
+                   windowsGraphicalUserInterface: false);
+
+                File.SetAttributes(sourceAppHostMock, FileAttributes.Normal);
+            }
+        }
+
         private string PrepareAppHostMockFile(TestDirectory testDirectory, Action<byte[]> customize = null)
         {
             // For now we're testing the AppHost on Windows PE files only.

